samcli/commands/_utils/options.py

```diff
@@ -0,0 +1,81 @@
+
+import os
+import click
+
+_TEMPLATE_OPTION_DEFAULT_VALUE = "template.[yaml|yml]"
+_TEMPLATE_SEARCH_PATHS = [
```

Comment:
Instead. What if you keep the default and have commands pass in the extra resolving that should take priority? 

This will simplify resolving and will allow the commands to control the resolve order. For example, package should only care about build and deploy should only care about package (templates). If they don't exist then some other flow would be triggered anyways.